Nodejs/Product/Nodejs/Intellisense/ExpansionClient.cs

```diff
@@ -0,0 +1,332 @@
+        private readonly ITextView _textView;
+        private readonly IVsEditorAdaptersFactoryService _adapterFactory;
+        private readonly IServiceProvider _serviceProvider;
+        private IVsExpansionSession _session;
+        private bool _sessionEnded, _selectEndSpan;
+        private ITrackingPoint _selectionStart, _selectionEnd;
+
+        public const string SurroundsWith = "SurroundsWith";
+        public const string SurroundsWithStatement = "SurroundsWithStatement";
+        public const string Expansion = "Expansion";
+
+        public ExpansionClient(ITextView textView, IVsEditorAdaptersFactoryService adapterFactory, IServiceProvider serviceProvider) {
+            _textView = textView;
+            _serviceProvider = serviceProvider;
+            _adapterFactory = adapterFactory;
+            _view = adapterFactory.GetViewAdapter(textView);
+            _lines = (IVsTextLines)adapterFactory.GetBufferAdapter(textView.TextBuffer);
+            _expansion = _lines as IVsExpansion;
+            if (_expansion == null) {
+                throw new ArgumentException("TextBuffer does not support expansions");
+            }
+        }
+
+        public bool InSession {
+            get {
+                return _session != null;
+            }
+        }
+
+        public int EndExpansion() {
+            _session = null;
+            _sessionEnded = true;
+            _selectionStart = _selectionEnd = null;
+            return VSConstants.S_OK;
+        }
+
+
+        public int FormatSpan(IVsTextLines pBuffer, TextSpan[] ts) {
+            IXMLDOMNode codeNode, snippetTypes, declarations, imports;
+
+            int hr;
+            if (ErrorHandler.Failed(hr = _session.GetSnippetNode("CodeSnippet:Code", out codeNode))) {
+                return hr;
+            }
+
+            if (ErrorHandler.Failed(hr = _session.GetHeaderNode("CodeSnippet:SnippetTypes", out snippetTypes))) {
+                return hr;
+            }
+
+            List<string> declList = new List<string>();
+            if (ErrorHandler.Succeeded(hr = _session.GetSnippetNode("CodeSnippet:Declarations", out declarations))
+                && declarations != null) {
+                foreach (IXMLDOMNode declType in declarations.childNodes) {
+                    var id = declType.selectSingleNode("./CodeSnippet:ID");
+                    if (id != null) {
+                        declList.Add(id.text);
+                    }
+                }
+            }
+
+            List<string> importList = new List<string>();
+            if (ErrorHandler.Succeeded(hr = _session.GetSnippetNode("CodeSnippet:Imports", out imports))
+                && imports != null) {
+                foreach (IXMLDOMNode import in imports.childNodes) {
+                    var id = import.selectSingleNode("./CodeSnippet:Namespace");
+                    if (id != null) {
+                        importList.Add(id.text);
+                    }
+                }
+            }

```

Comment:
You're not actually using Imports or importList anywhere. Do you have plans to? For instance, it might be useful with any snippets that use require